docs/reverse-replication/ReverseReplicationUserGuide.md

```diff
@@ -40,31 +56,45 @@ A few prerequisites must be considered before starting with reverse replication.
-9. [Source shards file](https://github.com/GoogleCloudPlatform/spanner-migration-tool/tree/master/reverse_replication#sample-sourceshards-file) already uploaded to GCS.
+8. Ensure that that [session file](./RunnigReverseReplication.md#files-generated-by-spanner-migration-tool) is uploaded to GCS (this requires a schema conversion to be done).
+9. [Source shards file](./RunnigReverseReplication.md#sample-sourceshards-file) already uploaded to GCS.
 10. Resources needed for reverse replication incur cost. Make sure to read [cost](#cost).
 
 ## Launching reverse replication
 
-Currently, the reverse replication flow is launched manually via a script. The details for the same are documented [here](https://github.com/GoogleCloudPlatform/spanner-migration-tool/tree/master/reverse_replication#reverse-replication-setup).
+Currently, the reverse replication flow is launched manually via a script. The details for the same are documented [here](./RunnigReverseReplication.md#reverse-replication-setup).
 
 ## Observe, tune and troubleshoot
 
 ### Tracking progress
 
-The progress of the Dataflow jobs can be tracked via the Dataflow UI.
+There are various progress points in the pipeline. Below sections detail how to track progress at each of them.
+
+#### Verify that change stream has data
+
+Unless there is change stream data to stream from Spanner, nothing will be reverse replicated. The first step is to verify that change stream has data. Refer [here](https://cloud.google.com/spanner/docs/change-streams/details#query) on how to check this.
+
 
 #### Metrics for Dataflow job that writes from Spanner to Sink
 
-The last step gives an approximation of where the step is currently - the Data Watermark would give indication of Spanner commit timestamp that is guaranteed to be processed.
+The progress of the Dataflow jobs can be tracked via the Dataflow UI.
+
+The last step gives an approximation of where the step is currently - the Data Watermark would give indication of Spanner commit timestamp that is guaranteed to be processed. On the Dataflow UI, click on JobGraph and scroll to the last step, as shown below. Click on the last step and the metrics should be visible on the right pane.
 
 ![Metrics](./images/SourceToSinkMetrics.png)
 
+
+#### Pub/Sub metrics
+
+Track the Pub/Sub topic [metrics](https://cloud.google.com/pubsub/docs/monitor-topic) to verify that there is inflow of messages by checking the 'Published Messages' metric.
+
+Note that subscription [metrics](https://cloud.google.com/pubsub/docs/monitor-subscription) can be verified to check if the dataflow job that writes to source database is reading from Pub/Sub as expected.
+
 #### Metrics for Dataflow job that writes to source database
 
-The Dataflow job that writes to source database exposes per shard metric like so:
+The Dataflow job that writes to source database exposes per shard metric like so, which should be visible on the right pane titled 'Job Info'.
```

Comment:
Please move the images out of the repo as it adds to bloat on the repo.

Reply:
Will move them once the images get approved

Reply:
Done, have removed the local images